sbom-utility-scripts/scripts/add-image-reference-script/add_image_reference.py

```diff
@@ -0,0 +1,404 @@
+    """
+    Find the package in the SBOM by SPDX ID.
+
+    Args:
+        sbom (dict): SBOM in JSON format.
+        spdx_id (str): SPDX ID of the package to find.
+
+    Returns:
+        dict: The package with the given SPDX ID.
+    """
+    for package in sbom["packages"]:
+        if package["SPDXID"] == spdx_id:
+            return package
+    return None
+
+
+def delete_package_by_spdx_id(sbom: dict, spdx_id: str) -> dict:
+    """
+    Delete the package in the SBOM by SPDX ID.
+
+    Args:
+        sbom (dict): SBOM in JSON format.
+        spdx_id (str): SPDX ID of the package to delete.
+
+    Returns:
+        dict: Updated SBOM with the package deleted.
+    """
+    for package in sbom["packages"]:
+        if package["SPDXID"] == spdx_id:
+            sbom["packages"].remove(package)
+            break
+    return sbom
+
+
+def redirect_virtual_root_to_new_root(sbom: dict, virtual_root: str, new_root: str) -> dict:
+    """
+    Redirect the relationship describing the document to a new root node.
+
+    Args:
+        sbom (dict): SBOM in JSON format.
+        virtual_root (str): A virtual root node identifier that needs to be replaced.
+        new_root (str): A new root node identifier that will replace the virtual root node.
+
+    Returns:
+        dict: Updated SBOM with the virtual root node replaced.
+    """
+    for relationship in sbom["relationships"]:
+        if relationship["spdxElementId"] == virtual_root:
+            relationship["spdxElementId"] = new_root
+
+
+def delete_relationship_by_related_spdx_id(sbom: dict, spdx_id: str) -> dict:
+    """
+    Delete the relationship in the SBOM by SPDX ID.
+
+    Args:
+        sbom (dict): SBOM in JSON format.
+        spdx_id (str): SPDX ID of the relationship to delete.
+
+    Returns:
+        dict: Updated SBOM with the relationship deleted.
+    """
+    for relationship in sbom["relationships"]:
+        if relationship["relatedSpdxElement"] == spdx_id:
+            sbom["relationships"].remove(relationship)
+            break
+    return sbom
+
+
+def describes_the_document(relationship_element: dict, doc_spdx_id: str) -> bool:
+    """
+    Check if the relationship describes the document.
+
+    Args:
+        relationship_element (dict): A relationship element from the SBOM that needs to be checked.
+        doc_spdx_id (str): A SPDX ID of the document.
+
+    Returns:
+        bool: A boolean indicating if the relationship describes the document.
+    """
+    return (
+        relationship_element["spdxElementId"] == doc_spdx_id and relationship_element["relationshipType"] == "DESCRIBES"
+    )
+
+
+def is_virtual_root(package: dict) -> bool:
+    """
+    Check if the package is a virtual - usually a package with empty values.
+
+    Args:
+        package (dict): A package element from the SBOM.
+
+    Returns:
+        bool: A boolean indicating if the package is a virtual root.
+    """
+    return package.get("name") == ""
```

Comment:
```suggestion
    return not package.get("name")
```
(for missing `name`, the current code would return False)

Reply:
Fixed